packages/server/lib/browsers/chrome.ts

```diff
@@ -580,11 +580,17 @@ export = {
   async connectToNewSpec (browser: Browser, options: BrowserNewTabOpts, automation: Automation) {
     debug('connecting to new chrome tab in existing instance with url and debugging port', { url: options.url })
 
-    const browserCriClient = this._getBrowserCriClient()
-
-    if (!browserCriClient) throw new Error('Missing browserCriClient in connectToNewSpec')
+    let browserClient = this._getBrowserCriClient();
+
+    if (!browserClient) {
+      // create a new instance of the client in case something went wrong
+      // https://github.com/cypress-io/cypress/issues/24650
+      const port = await protocol.getRemoteDebuggingPort()
+      browserCriClient = await BrowserCriClient.create(['127.0.0.1'], port, browser.displayName, options.onError, onReconnect)
```

Comment:
since we use the same connection logic elsewhere I wonder if we can refactor this. @ryanthemanuel is this something we should be doing?

Reply:
I was thinking the same. I kind of wanted to get some discussion on this so we could try and attempt to fix this issue which is causing us some major head-aches. I'm more than happy to contribute towards the fix if need be.